core/container/src/main/java/org/wildfly/swarm/Swarm.java

```diff
@@ -464,7 +464,13 @@ private static boolean isFatJar() throws IOException {
 
     private void loadStageConfiguration(URL url) {
         List<ProjectStage> projectStages = new ProjectStageFactory().loadStages(url);
-        String stageName = System.getProperty(SwarmProperties.PROJECT_STAGE, "default");
+        String stageName = System.getProperty(SwarmProperties.PROJECT_STAGE);
```

Comment:
Not saying that Optional isn't a good approach, I just didn't want to spend much time on rewriting the whole method to use Optional (specially considering it does not belong in this Sprint 😜)